stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -5338,7 +5338,16 @@ internal enum _Variant${Self}Buffer<${TypeParametersDecl}> : _HashBuffer {
 
     var (i, found) = asNative._find(key, startBucket: asNative._bucket(key))
     if found {
-      _ = ensureUniqueNativeBuffer(asNative.capacity)
+      // This is a performance optimization that was put in to ensure that we
+      // did not make a copy of self to call asNative.capacity over
+      // ensureUniqueNativeBefore causing at -Onone the uniqueness check to
+      // fail. This code used to be:
+      //
+      // _ = ensureUniqueNativeBuffer(asNative.capacity)
+      //
+      // SR-6437
+      let capacity = asNative.capacity
+      _ = ensureUniqueNativeBuffer(capacity)
```

Comment:
Shouldn't we just call _nativePointerToValue(at:) here?

Reply:
Thanks @lorentey, I've gone ahead and made that change.